mrp_bom_nested_bom_generator/models/mrp_bom.py

```diff
@@ -0,0 +1,154 @@
+from odoo import _, api, fields, models
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    nested_bom = fields.Boolean(string="Nested BOM")
+    nested_bom_updated = fields.Boolean(
+        string="Changed Nested Bom",
+        compute="_compute_nested_bom_updated",
+        store=True,
+        default=True,
+    )
+
+    nested_bom_ids = fields.One2many(
+        comodel_name="mrp.nested.bom",
+        inverse_name="bom_id",
+    )
+
+    parent_bom_id = fields.Many2one(comodel_name="mrp.bom", string="Parent MRP BOM")
+
+    child_bom_ids = fields.One2many(
+        comodel_name="mrp.bom", inverse_name="parent_bom_id", string="Child BOMs"
+    )
+
+    CHOICE_FUNC = {True: "_create_mrp_bom_record", False: "_append_bom_line_components"}
```

Comment:
What is the reason of doing it this way? Are there any chances of implementing new functions instead of extending existing ones?

Reply:
Resolved